gearcmd/worker.go

```diff
@@ -39,10 +45,40 @@ var (
 	legacyLg = logger.New("gearman")
 )
 
+// ProcessWithErrorBackoff calls Process and sleeps if the last N jobs returned an error
+func (conf *TaskConfig) ProcessWithErrorBackoff(job baseworker.Job) (b []byte, returnErr error) {
+	b, returnErr = conf.Process(job)
+	if conf.LastResults == nil || conf.ErrorResultsBackoffRate == 0 {
+		return b, returnErr
+	}
+
+	conf.LastResults.Value = returnErr
+	conf.LastResults = conf.LastResults.Next()
+
+	allErrors := true
```

Comment:
can we make this a more descriptive? it wasn't obvious what was going on at first